paramiko/agent.py

```diff
@@ -109,9 +109,12 @@ def __init__(self, agent):
     def run(self):
         try:
             (r, addr) = self.get_connection()
+            # Found that r should be either a socket from the socket library or None
             self.__inr = r
-            self.__addr = addr
+            self.__addr = addr # This should be an IP address as a string? or None
             self._agent.connect()
+            if self._agent._conn is None or not (hasattr(self._agent._conn, 'fileno') or isinstance(self._agent, int)):
```

Comment:
@bitprophet I was trying to find my original traceback message, but the original from #402 is close enough. The very last line mentions that it's looking for it be an ```int``` object or have the method. So I added in the either/or test because both of those tests need to be ```False``` in order to avoid the traceback apparently. I forget why I put the check for the ```None``` type first. 

Anyway, if you still want me to change it I will.